### PR TITLE
docs: add steps to local dev guide

### PR DESCRIPTION
**Pull request by njelich**
Based on my experience as a non PHP developer, the guide to work on this locally is nowhere near sufficient to actually get started.

Features such as the dummy data generation are tucked away while they are extremely useful for testing the app and the setup of the database is not initially clear.

Based on that I've expanded and rewritten the steps a bit. Also added `yarn start` which starts both vite and php artisan serve, since the ergonomics is better that way (and I originally lost quite some time figuring out I need to php artisan serve for anything at all to work).

Please squash merge when merging.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,20 @@
-## Run locally
-
-1. `composer install`
-1. `cp .env.example .env` and configure `.env` file
-1. `yarn install`
-1. `yarn run dev`
-1. Optional: make the search work:
+## Local development
+
+1. Install [Valet](https://laravel.com/docs/9.x/valet)
```
Comment (djaiss) [inline README.md]: We should indicate in the first step that we need PHP to work locally as well as a web server. We should also say that, for Mac owners, the preferred way is Valet. 

Most of the contributors of the old codebase don't have MacOS.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+2. Install [sqlite](https://formulae.brew.sh/formula/sqlite)
```
Comment (djaiss) [inline README.md]: Here we should say that they need a relational database, that we support MySQL and SQLite but we recommend sqlite.

File: README.md [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+3. `composer install --no-progress --no-interaction --prefer-dist --optimize-autoloader`
+4. `yarn install --frozen-lockfile`
+5. `cp .env.example .env` and configure `.env` file
+   1. `php artisan key:generate --no-interaction` (generates APP_KEY)
+   2. `touch monica.db` and add path to DB_DATABASE
```
Comment (djaiss) [inline README.md]: IF we use sqlite, otherwise instructions are different. For instance, if they use Homestead.

File: package.json [hunk 4]
```diff
@@ -3,6 +3,7 @@
   "scripts": {
     "dev": "vite --https",
     "build": "vite build",
+    "start": "vite --https & php artisan serve && fg",
```
Comment (djaiss) [inline package.json]: I personally don't need this step. `yarn dev` is enough on my machine. What's `fg`?

**Review by djaiss — CHANGES_REQUESTED** [review]
Thanks a lot for contributing. Your changes are welcome and needed.

File: README.md (hunk 3 above)
Comment (njelich) [reply README.md]: I'm not familiar with Homestead, can you draft them up. This is the one happy path I managed to explore.

File: package.json (hunk 4 above)
Comment (njelich) [reply package.json]: If I am not running artisan serve nothing works for me.

When I run vite it opens the vite page, but doesn't actually provide me with any frontend. The frontend for me shows up on the localhost:8000 page that php artisan serve spits out.

File: README.md (hunk 2 above)
Comment (njelich) [reply README.md]: Yep, I went with the happy path that worked for me. Sqlite is a lot easier and simpler tho - and for local development in particular is the best option (no server to run).

File: README.md (hunk 1 above)
Comment (njelich) [reply README.md]: Can you explain that please, not familiar enough with PHP to get into the nuances.

File: package.json (hunk 4 above)
Comment (njelich) [reply package.json]: Oh god, I truly am not the guy to mess around with php. Yep. Works fine on my side with just yarn dev as well. But it starts on port 8000, which is the same as the default of php artisan serve. Hence, I though the php artisan serve was making it work, while really the APP_URL env var should be updated.

File: README.md (hunk 2 above)
Comment (djaiss) [reply README.md]: You are right.

**Comment by djaiss** [comment]
@njelich thanks for the changes!
Also, we always always squash commits :-)

**Review by djaiss — APPROVED** [review]

**Pull request merged**